src/noisepy/seis/zarrstore.py

```diff
@@ -0,0 +1,154 @@
+import datetime
+import glob
+import logging
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import obspy
+import zarr
+from datetimerange import DateTimeRange
+
+from . import noise_module
+from .constants import DATE_FORMAT, DONE_PATH, PROGRESS_DATATYPE
+from .datatypes import Channel, ChannelData, ChannelType, ConfigParameters, Station
+from .stores import CrossCorrelationDataStore, RawDataStore
+
+logger = logging.getLogger(__name__)
+
+
+class ZarrRawDataStore(RawDataStore):
+    """
+    A data store implementation to read from a directory of Zarr files. Each file is considered
+    a timespan with the naming convention: 2019_02_01_00_00_00T2019_02_02_00_00_00.zarr
+    """
+
+    def __init__(self, directory: str):
+        super().__init__()
+        self.directory = directory
+        zarr_files = sorted(glob.glob(os.path.join(directory, "*.zarr")))
+        self.files = {str(self._parse_timespans(f)): f for f in zarr_files}
+        logger.info(f"Initialized store with {len(self.files)}")
+
+    def get_channels(self, timespan: DateTimeRange) -> List[Channel]:
+        store = zarr.DirectoryStore(self.files[str(timespan)])
+        arr = zarr.open(store)
+        # Assuming your Zarr structure follows a similar pattern, you may need to adjust the following lines accordingly
+        stations = [self._create_station(arr, sta) for sta in arr["waveforms"].keys() if sta is not None]
+        channels = [Channel(ChannelType(tag), sta) for sta in stations for tag in arr["waveforms"][str(sta)].keys()]
+        return channels
+
+    def get_timespans(self) -> List[DateTimeRange]:
+        return [DateTimeRange.from_range_text(d) for d in sorted(self.files.keys())]
+
+    def read_data(self, timespan: DateTimeRange, chan: Channel) -> np.ndarray:
+        store = zarr.DirectoryStore(self.files[str(timespan)])
+        arr = zarr.open(store)
+        stream = arr["waveforms"][str(chan.station)][str(chan.type)][0]
+        return ChannelData(stream, arr.attrs["sampling_rate"], arr.attrs["starttime"])
+
+    def _parse_timespans(self, filename: str) -> DateTimeRange:
+        parts = os.path.splitext(os.path.basename(filename))[0].split("T")
+        dates = [obspy.UTCDateTime(p).datetime.replace(tzinfo=datetime.timezone.utc) for p in parts]
+        return DateTimeRange(dates[0], dates[1])
+
+    def _create_station(self, arr, name: str) -> Optional[Station]:
+        # What should we do if there's not StationXML?
+        try:
+            inventory = arr[name]["StationXML"]
+            sta, net, lon, lat, elv, loc = noise_module.sta_info_from_inv(inventory)
+            return Station(net, sta, lat, lon, elv, loc)
+        except Exception as e:
+            logger.warning(f"Missing StationXML for station {name}. {e}")
+            return None
+
+
+class ZarrCCStore(CrossCorrelationDataStore):
```

Comment:
Please add unit testing for this class. Take a look at `test_asdfstore.py:test_ccstore`. I think we can reuse the same test code for both the ASDF and Zarr CC stores. I.e. you can make a helper test method that takes an abstract `CrossCorrelationStore` as an argument and then write two tests that call it with different stores. E.g.
```

def _ccstore_test_helper(ccstore: CrossCorrelationStore):
... current test code

def test_asdfccstore(asdfstore: ASDFCCStore):
    _ccstore_test_helper(asdfstore)

def test_zarrccstore(zarrstore: ZarrCCStore):
    _ccstore_test_helper(zarrstore)
```
You would need to add a new test fixture like `ccstore` but for Zarr.